src/script/descriptor.cpp

```diff
@@ -464,9 +464,11 @@ std::unique_ptr<PubkeyProvider> ParsePubkey(const Span<const char>& sp, bool per
             return MakeUnique<ConstPubkeyProvider>(pubkey);
         }
     }
-    CExtKey extkey = DecodeExtKey(str);
-    CExtPubKey extpubkey = DecodeExtPubKey(str);
-    if (!extkey.key.IsValid() && !extpubkey.pubkey.IsValid()) return nullptr;
+    CExtKey extkey;
+    bool validExtKey = DecodeExtKey(str, extkey);
+    CExtPubKey extpubkey;
+    bool validExtPubKey = DecodeExtPubKey(str, extpubkey);
+    if (!(validExtKey && extkey.key.IsValid()) && !(validExtPubKey && extpubkey.pubkey.IsValid())) return nullptr;
```

Comment:
Why do you check `IsValid` here again?  That's already contained in the return value you added.  So either you should make it explicit in `DecodeExtKey`'s contract that the returned key is valid if `true` is returned and remove the check here (preferred for me), or you should not check the key for validity in `DecodeExtKey` and have the callers check it - but not both.